fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -532,9 +538,42 @@ public void abortTransaction(long transactionId, String reason, TxnCommitAttachm
             writeUnlock();
             transactionState.afterStateTransform(TransactionStatus.ABORTED, txnOperated, reason);
         }
+
+        // send clear txn task to BE to clear the transactions on BE.
+        // This is because parts of a txn may succeed in some BE, and these parts of txn should be cleared
+        // explicitly, or it will be remained on BE forever
+        // (However the report process will do the diff and send clear txn tasks to BE, but that is our
+        // last defense)
+        if (txnOperated && transactionState.getTransactionStatus() == TransactionStatus.ABORTED) {
+            clearBackendTransactions(transactionState);
+        }
+
         return;
     }
     
+    private void clearBackendTransactions(TransactionState transactionState) {
+        Preconditions.checkState(transactionState.getTransactionStatus() == TransactionStatus.ABORTED);
+        // for aborted transaction, we don't know which backends are involved, so we have to send clear task
+        // to all backends.
+        List<Long> allBeIds = Catalog.getCurrentSystemInfo().getBackendIds(false);
+        synchronized (clearTransactionTasks) {
+            for (Long beId : allBeIds) {
+                ClearTransactionTask task = new ClearTransactionTask(beId, transactionState.getTransactionId(), null);
+                clearTransactionTasks.add(task);
+            }
+
+            // try to group send tasks, not sending every time a txn is aborted. to avoid too many task rpc.
+            if (clearTransactionTasks.size() > allBeIds.size() * 2) {
+                AgentBatchTask batchTask = new AgentBatchTask();
+                for (ClearTransactionTask clearTransactionTask : clearTransactionTasks) {
+                    batchTask.addTask(clearTransactionTask);
+                }
+                AgentTaskExecutor.submit(batchTask);
```

Comment:
why not submit task out of synchronized block?

Reply:
Just want one thread to submit the clear task, or if there are lots of aborted txn, there will be
lots of clear tasks being submitted concurrently

Reply:
Emm, what I suggest is that you construct batchTask in the synchronized code block, and then you can call AgentTaskExecutor.submit(batchTask). Even though `AgentTaskExecutor.submit` is async now, it doesn't means it won't be changed in the future. And others may not known every detail when he want to change some thing. So there is no harm to do this.